packages/connect-migrate/src/migrate.ts

```diff
@@ -19,50 +19,179 @@ import { run } from "./run";
-    "@connectrpc/protoc-gen-connect-query-react",
+  "@bufbuild/connect": {
+    newPackage: "@connectrpc/connect",
+    version: "0.13.1",
+  },
+  "@bufbuild/connect-web": {
+    newPackage: "@connectrpc/connect-web",
+    version: "0.13.1",
+  },
+  "@bufbuild/connect-fastify": {
+    newPackage: "@connectrpc/connect-fastify",
+    version: "0.13.1",
+  },
+  "@bufbuild/connect-node": {
+    newPackage: "@connectrpc/connect-node",
+    version: "0.13.1",
+  },
+  "@bufbuild/connect-next": {
+    newPackage: "@connectrpc/connect-next",
+    version: "0.13.1",
+  },
+  "@bufbuild/connect-express": {
+    newPackage: "@connectrpc/connect-express",
+    version: "0.13.1",
+  },
+  "@bufbuild/protoc-gen-connect-es": {
+    newPackage: "@connectrpc/protoc-gen-connect-es",
+    version: "0.13.1",
+  },
+  "@bufbuild/protoc-gen-connect-web": {
+    newPackage: "@connectrpc/protoc-gen-connect-web",
+    version: "0.13.1",
+  },
+  "@bufbuild/connect-query": {
+    newPackage: "@connectrpc/connect-query",
+    version: "0.4.2",
+  },
+  "@bufbuild/protoc-gen-connect-query": {
+    newPackage: "@connectrpc/protoc-gen-connect-query",
+    version: "0.4.2",
+  },
+  "@bufbuild/protoc-gen-connect-query-react": {
+    newPackage: "@connectrpc/protoc-gen-connect-query-react",
+    version: "0.4.2",
+  },
 } as const;
 
 const keys = Object.keys(replacementMap) as (keyof typeof replacementMap)[];
 
+interface UsedPackage {
+  packageName: keyof typeof replacementMap;
+  version: string;
+  location: "dependencies" | "devDependencies" | "peerDependencies";
+}
+
 export function getReplacementImport(sourceValue: string): string | undefined {
   const match = keys.find((key) => {
     return sourceValue === key || sourceValue.startsWith(`${key}/`);
   });
   if (sourceValue === match) {
-    return replacementMap[match];
+    return replacementMap[match].newPackage;
   } else if (match !== undefined) {
-    return sourceValue.replace(`${match}/`, `${replacementMap[match]}/`);
+    return sourceValue.replace(
+      `${match}/`,
+      `${replacementMap[match].newPackage}/`,
+    );
   }
   return undefined;
 }
 
-export function replacePackageJSONReferences(jsonString: string): string {
+export function replacePackageJSONReferences(
+  jsonString: string,
+  packagesToForceUpdate?: { packageName: string }[],
+): string {
   let result = jsonString;
   for (const key of keys) {
-    result = result.replace(`"${key}"`, `"${replacementMap[key]}"`);
+    const forceUpdate =
+      packagesToForceUpdate?.find((p) => p.packageName === key) !== undefined;
+    if (!forceUpdate) {
+      result = result.replace(
+        `"${key}"`,
+        `"${replacementMap[key].newPackage}"`,
+      );
+    } else {
+      // I wish there was a way to do this without relying on dynamic regexes
+      // but not while maintaining existing formatting
+      result = result.replace(
+        new RegExp(`"${key}"(\\s*:\\s*)"\\S*"([\\s,])`, "g"),
```

Comment:
If we're willing to rewrite the entire package.json file, we can simplify here, but I'm not sure we want to do that.

Reply:
Pushed up https://github.com/connectrpc/connect-es/pull/782/commits/315540a88b560d75af9ba4c722dc75d0b042b9bc to go through JSON, formatting with two spaces for indentation. We run `npm install` too, so I think that's pretty reasonable.